src/main/java/ch/njol/skript/expressions/ExprFishingHook.java

```diff
@@ -0,0 +1,57 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.lang.ExpressionType;
+import org.bukkit.entity.FishHook;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Fishing Hook")
+@Description("The <a href='classes.html#entity'>fishing hook</a> in a fishing event.")
+@Examples({
+	"on fishing:",
+		"\twait a second",
+		"\tteleport player to fishing hook"
+})
+@Events("Fishing")
+@Since("INSERT VERSION")
+public class ExprFishingHook extends EventValueExpression<FishHook> {
```

Comment:
You're already registering an event value expression for FishHook in the ClassInfo of FishHook. I don't believe this class is needed.